File: frontend/src/types/agent.ts
export interface Agent {
  id: string;
  name: string;
  description: string;
  icon: string;
  category: string;
  purpose?: string;
  tags?: string[];
  integrations?: string[];
  is_favorite: boolean;
  usage_count: number;
  last_used: string | null;
  last_generated?: string | null;
}

export interface AgentListResponse {
  agents: Agent[];
}

export interface ConversationMode {
  id: string;
  label: string;
  description: string;
}

export interface AgentDetailResponse {
  agent: Agent;
  modes: ConversationMode[];
}

export const AGENT_CATEGORIES = [
  "People Management",
  "Leadership & Strategy",
  "Productivity & Efficiency",
] as const;

export type AgentCategory = (typeof AGENT_CATEGORIES)[number];

File: backend/app/schemas/agent.py
from typing import Optional
from pydantic import BaseModel


class AgentInfo(BaseModel):
    id: str
    name: str
    description: str
    icon: str
    category: str
    purpose: Optional[str] = None
    tags: list[str] = []
    integrations: list[str] = []
    is_favorite: bool
    usage_count: int
    last_used: Optional[str] = None
    last_generated: Optional[str] = None


class AgentListResponse(BaseModel):
    agents: list[AgentInfo]


class ConversationMode(BaseModel):
    id: str
    label: str
    description: str


class AgentDetailResponse(BaseModel):
    agent: AgentInfo
    modes: list[ConversationMode]
